common/buildcraft/factory/tile/TileMiningWell.java

```diff
@@ -1,74 +1,25 @@
 package buildcraft.factory.tile;
 
-import java.io.IOException;
-import java.util.List;
-
-import net.minecraft.block.state.IBlockState;
-import net.minecraft.item.ItemStack;
-import net.minecraft.nbt.NBTTagCompound;
-import net.minecraft.network.PacketBuffer;
-import net.minecraft.util.EnumFacing;
-import net.minecraft.util.ITickable;
-import net.minecraft.util.math.BlockPos;
-import net.minecraft.world.WorldServer;
-
-import net.minecraftforge.common.MinecraftForge;
-import net.minecraftforge.event.world.BlockEvent;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
-
-import buildcraft.api.mj.MjBattery;
-import buildcraft.api.tiles.IControllable;
-import buildcraft.api.tiles.IDebuggable;
-import buildcraft.api.tiles.IHasWork;
 import buildcraft.core.lib.inventory.InvUtils;
 import buildcraft.core.lib.utils.BlockUtils;
 import buildcraft.core.lib.utils.Utils;
 import buildcraft.factory.BCFactoryBlocks;
-import buildcraft.lib.migrate.BCVersion;
 import buildcraft.lib.misc.FakePlayerUtil;
-import buildcraft.lib.misc.MessageUtil;
-import buildcraft.lib.tile.TileBC_Neptune;
-
-public class TileMiningWell extends TileBC_Neptune implements ITickable, IHasWork, IControllable, IDebuggable {
-    public static final int NET_LED_STATUS = 10;
-
-    private int progress = 0;
-    private BlockPos currentPos = null;
-    // Used to check if this has completed all work
-    private boolean isComplete = false;
-    private Mode mode = Mode.On;
-    private final MjBattery battery = new MjBattery(1000_000);
+import net.minecraft.block.Block;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.init.Blocks;
+import net.minecraft.item.ItemStack;
+import net.minecraft.world.WorldServer;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.event.world.BlockEvent;
 
-    private void initCurrentPos() {
-        if (currentPos == null) {
-            currentPos = getPos();
-            currentPos = currentPos.down();
-        }
-    }
+import java.util.List;
 
+public class TileMiningWell extends TileMiner {
     @Override
-    public void update() {
-        if (worldObj.isRemote || mode != Mode.On) {
-            return;
-        }
-        battery.tick(getWorld(), getPos());
-
-        // test with the output of a stone engine
-        battery.addPower(1000);// remove this
-
-        if (worldObj.rand.nextDouble() > 0.9) {
-            sendNetworkUpdate(NET_LED_STATUS);
-        }
-
-        if (isComplete) {
-            return;
-        }
-
-        initCurrentPos();
-
+    protected void mine() {
         IBlockState state = worldObj.getBlockState(currentPos);
-        if (BlockUtils.isUnbreakableBlock(getWorld(), currentPos)) {
+        if (BlockUtils.isUnbreakableBlock(getWorld(), currentPos) || state.getBlock() == Blocks.BEDROCK) {
```

Comment:
It looks like BlockUtils.isUnbreakableBlock is failing to identify bedrock as unbreakable, ideally you would not have to also check here.